### PR TITLE
perf(rpc): use `Arc<SealedBlockWithSenders>` on cache and rpc

### PR DESCRIPTION
**Pull request by joshieDo**
* Use it on the `full_block_cache` LRU instead of `Arc<BlockWithSenders>`
* Cache should not act as a provider for types it doesn't store and would need to be calculated.  
  * Removes `CacheAction::GetBlockTransactions ` since it was fetching `Arc<BlockWithSenders>` anyway.
  * Removes a bunch of methods that were just getting `Arc<BlockWithSenders>` and cloning prematurely. There are places that a reference is enough
* There are still some places that need some changes to avoid cloning the block, but the PR is already big enough...

### CODE REVIEW

**Review by emhane — APPROVED** [review]

File: crates/rpc/rpc-eth-api/src/helpers/transaction.rs [hunk 1]
```diff
@@ -270,19 +277,21 @@ pub trait EthTransactions: LoadTransaction {
...
+                        .find(|(_, (signer, tx))| **signer == sender && tx.nonce() == nonce)
+                        .map(|(index, (signer, tx))| {
                             let tx_info = TransactionInfo {
                                 hash: Some(tx.hash()),
                                 block_hash: Some(block_hash),
                                 block_number: Some(block_number),
                                 base_fee: base_fee_per_gas.map(u128::from),
                                 index: Some(index as u64),
                             };
+                            // TODO(joshie) rpc compat should accept a reference
```
[the remark this answers is not in the document]
Comment (emhane) [reply crates/rpc/rpc-eth-api/src/helpers/transaction.rs:291]: actually, better name for `TransactionCompat::fill`, is `into`. and adding a second method which takes a ref to tx, should be called `to`.

File: crates/rpc/rpc/src/debug.rs [hunk 2]
```diff
@@ -219,7 +219,7 @@ where
 
         self.trace_block(
             state_at.into(),
-            block.into_transactions_ecrecovered().collect(),
+            (*block).clone().into_transactions_ecrecovered().collect(),
```
Comment (mattsse) [inline crates/rpc/rpc/src/debug.rs:222]: this will also clone the other body components, but is fine here

**Review by mattsse — APPROVED** [review]
all of this makes sense.

separating the address from the tx is also a good change because this makes it easier to deal with different layers of tx types and we also want to convert this into some kind of associated trait eventually

File: crates/rpc/rpc-eth-api/src/helpers/transaction.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (mattsse) [reply crates/rpc/rpc-eth-api/src/helpers/transaction.rs]: I like that yes

**Pull request merged**